### PR TITLE
build(deps): bump slackapi/slack-github-action from 1.25.0 to 2.0.0

### LINKED ISSUE #16749: update workflow to function with major slack version roll
New major version: https://github.com/slackapi/slack-github-action/releases/tag/v2.0.0

Rolling from version 1.25.0 to 2.0.0 broke slack's functions. This needs to be resolved.

Here's the originating PR:
https://github.com/hashgraph/hedera-services/pull/16611

Here's the revert PR:
https://github.com/hashgraph/hedera-services/pull/16747

And here's an example of the workflow failing:
https://github.com/hashgraph/hedera-services/actions/runs/11965768181/job/33361748643
Step: `Report XTS execution failure` > `Report failure (slack`

### PR DESCRIPTION
**Pull request by PavelSBorisov**
**Description**:
This PR updates the slack-github-action to the newest major version [https://github.com/slackapi/slack-github-action/releases/tag/v2.0.0](https://github.com/slackapi/slack-github-action/releases/tag/v2.0.0)
Implements the new variable to allow templated payloads which were enabled by default in the previous version - [https://github.com/slackapi/slack-github-action/releases/tag/v2.0.0#only-one-payload-input-can-be-provided](https://github.com/slackapi/slack-github-action/releases/tag/v2.0.0#only-one-payload-input-can-be-provided)
**Related issue(s)**:

Fixes #16749

**Notes for reviewer**:
N/A

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by rbarkerSL** [comment]
I ran a test against this branch on prepare-xts-candidate: https://github.com/hashgraph/hedera-services/actions/runs/12035146720

It failed because the incoming webhook isn't set correctly. I believe that it wants `SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK` to be `SLACK_WEBHOOK_TYPE: 'incoming-webhook'`

File: .github/workflows/zxcron-promote-build-candidate.yaml
Comment (rbarkerSL) [inline .github/workflows/zxcron-promote-build-candidate.yaml:1]: This file has a newly added section: `report-promotion:` which will need the same updates as well. It's at line 152

```
      - name: Report Promoted Build
        uses: slackapi/slack-github-action@6c661ce58804a1a20f6dc5fbee7f0381b469e001 # v1.25.0
        env:
          SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_CITR_OPERATIONS_WEBHOOK }}
        with:
          payload: |
```
becomes
```
      - name: Report Promoted Build
        uses: slackapi/slack-github-action@485a9d42d3a73031f12ec201c457e2162c45d02d # v2.0.0
        with:
          payload-templated: true
          webhook: ${{ secrets.SLACK_CITR_OPERATIONS_WEBHOOK }}
          webhook-type: incoming-webhook
          payload: |
```

**Comment by PavelSBorisov** [comment]
> I ran a test against this branch on prepare-xts-candidate: https://github.com/hashgraph/hedera-services/actions/runs/12035146720
>It failed because the incoming webhook isn't set correctly. I believe that it wants SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK to be SLACK_WEBHOOK_TYPE: 'incoming-webhook'

Ran another test after fixing the above, result seems to be reported correctly now:
https://github.com/hashgraph/hedera-services/actions/runs/12067628115/job/33651001636

**Review by rbarkerSL — APPROVED** [review]
Looks great

**Pull request merged**